crates/swc_ecma_transforms_module/src/system_js.rs

```diff
@@ -0,0 +1,1189 @@
+                                expr = Expr::Call(self.export_call(
+                                    value.clone(),
+                                    DUMMY_SP,
+                                    expr.as_arg(),
+                                ));
+                            }
+                            return Expr::Seq(SeqExpr {
+                                span: DUMMY_SP,
+                                exprs: vec![Box::new(expr), Box::new(Expr::Update(update_expr))],
+                            });
+                        }
+                    }
+                    Expr::Update(update_expr)
+                }
+                _ => Expr::Update(update_expr),
+            }
+        } else {
+            Expr::Update(update_expr)
+        }
+    }
+
+    fn add_export_name(&mut self, key: Id, value: JsWord) {
+        let mut find = false;
+        for (k, v) in self.export_map.iter_mut() {
+            if key.eq(k) {
+                v.push(value.clone());
+                find = true;
+                break;
+            }
+        }
+        if !find {
+            self.export_map.insert(key, vec![value]);
+        }
+    }
+
+    fn add_declare_var_idents(&mut self, ident: &Ident) {
+        self.declare_var_idents.push(ident.clone());
+    }
+
+    fn build_export_call(
+        &mut self,
+        export_names: &mut Vec<JsWord>,
+        export_values: &mut Vec<Box<Expr>>,
+    ) -> Vec<Stmt> {
+        match export_names.len() {
+            0 => vec![],
+            1 => vec![self
+                .export_call(
+                    export_names.remove(0),
+                    DUMMY_SP,
+                    (*export_values.remove(0)).as_arg(),
+                )
+                .into_stmt()],
+            _ => {
+                let mut props = vec![];
+                while !export_names.is_empty() {
+                    let sym = export_names.remove(0);
+
+                    props.push(PropOrSpread::Prop(Box::new(Prop::KeyValue(KeyValueProp {
+                        key: match Ident::verify_symbol(&sym).is_ok() {
+                            true => PropName::Ident(quote_ident!(sym)),
+                            false => PropName::Str(quote_str!(sym)),
+                        },
+                        value: export_values.remove(0),
+                    }))));
+                }
+                return vec![CallExpr {
+                    span: DUMMY_SP,
+                    callee: Callee::Expr(Box::new(Expr::Ident(quote_ident!("_export")))),
+                    args: vec![Expr::Object(ObjectLit {
+                        span: DUMMY_SP,
+                        props,
+                    })
+                    .as_arg()],
+                    type_args: None,
+                }
+                .into_stmt()];
+            }
+        }
+    }
+
+    fn build_module_item_export_all(&mut self, mut meta: ModuleItemMeta) -> Vec<Stmt> {
+        match meta.has_export_all {
```

Comment:
I think `if !meta.has_export_all {} else {}` is better, as it's a dedicated syntax.